test/gpu/quantization.cpp

```diff
@@ -110,7 +111,16 @@ TEST_CASE(int8_quantization)
         migraphx::target gpu_t = migraphx::make_target("gpu");
         run_prog(p, gpu_t, m, gpu_result);
 
-        EXPECT(migraphx::verify_range(ref_result, gpu_result));
+        // Note: the tolerance for mlir_enabled result is temporarily bumped
+        // higher because the lowering pipeline between mlir fallback and
+        // regular non-mlir pipeline diverged. MLIR fallback uses the
+        // rewrite_quantization at the very end of the pipeline, whereas
+        // the regular pipeline uses the rewrite_quantization in the much
+        // earlier stage.
+        if(migraphx::gpu::mlir_enabled())
+            EXPECT(migraphx::verify_range(ref_result, gpu_result, 1e5));
```

Comment:
This isn't comparing to MLIR. This is comparing from
 - a. Quantization that happens normally, in very front of the pipeline
 - b. MLIR's fusion didn't capture the pointwise module, so quantization happens in fallback mode, in the very back of the pipeline.

In both cases, the quantizelinear and dequantizelinear was lowered within migraphx implementation. The error increased because the sequence matters. Effectively, in the fallback path, because of the sequence difference, it becomes the comparison between two possible lowering sequence of migraphx.

---

That's also why I'm trying to make migraphx support int32 type natively. Only if it does support int32 type, will it not populate redundant convert ops. And only when it doesn't populate redundant convert ops can MLIR lowering of quantizelinear and dequantizelinear happen at the front the pipeline. Then we can do the error comparison fairly.

Reply:
Ok I see. It does make sense to use a higher tolerance since this is using int8 internally and no fp32. I would rather use the same tolerance for both paths instead of making it conditional. We can lower it again when we get the int32 working.

Reply:
Sounds good, I'd prefer to just keep as-is if is it not a strong preference :-). I have double checked that once in32 type is enabled, this test will be able to pass as-is... So this is really just throw-away code in the next PR either one way or another.